contracts/nois-proxy/src/contract.rs

```diff
@@ -190,16 +214,76 @@ pub fn execute_get_randomness_impl(
     Ok(res)
 }
 
+#[allow(clippy::too_many_arguments)]
+fn execute_set_config(
+    deps: DepsMut,
+    info: MessageInfo,
+    env: Env,
+    manager: Option<String>,
+    prices: Option<Vec<Coin>>,
+    payment: Option<String>,
+    nois_beacon_price: Option<Uint128>,
+    mode: Option<OperationalMode>,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+
+    // if manager set, check the calling address is the authorised multisig
+    if let Some(manager) = config.manager.clone() {
+        ensure_eq!(info.sender, manager, ContractError::Unauthorized)
+    }
```

Comment:
This (and all other manager checks in this contract) should raise Unauthorized when there is no manager in the config. The current code just skips the check when no manager is set.

I played around with some ways to write that and this is the best way I found for now:

```rust
    ensure_eq!(
        &info.sender, // &Addr
        config.manager.as_ref().ok_or(ContractError::Unauthorized)?, // &Addr
        ContractError::Unauthorized
    );
```